modules/core/04-channel/keeper/keeper_test.go

```diff
@@ -541,3 +542,323 @@ func (suite *KeeperTestSuite) TestUnsetParams() {
+
+		// postPruneExpState is a helper function to verify the expected state after pruning. Argument expLeft
+		// denotes the expected amount of packet acks and receipts left after pruning. Argument expSequenceStart
+		// denotes the expected value of PruneSequenceStart.
+		postPruneExpState = func(expAcksLen, expReceiptsLen, expPruningSequenceStart uint64) {
+			acks := suite.chainA.App.GetIBCKeeper().ChannelKeeper.GetAllPacketAcks(suite.chainA.GetContext())
+			suite.Require().Len(acks, int(expAcksLen))
+
+			receipts := suite.chainA.App.GetIBCKeeper().ChannelKeeper.GetAllPacketReceipts(suite.chainA.GetContext())
+			suite.Require().Len(receipts, int(expReceiptsLen))
+
+			start := suite.chainA.App.GetIBCKeeper().ChannelKeeper.GetPruningSequenceStart(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+			suite.Require().Equal(start, expPruningSequenceStart)
+		}
+	)
+
+	testCases := []struct {
+		name     string
+		pre      func()
+		malleate func()
+		post     func(pruned, left uint64)
+		expError error
+	}{
+		{
+			"success: no packets sent, no stale packet state pruned",
+			func() {},
+			func() {},
+			func(pruned, left uint64) {
+				// Assert that PruneSequenceStart and PruneSequenceEnd are both set to 1.
+				start := suite.chainA.App.GetIBCKeeper().ChannelKeeper.GetPruningSequenceStart(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+				end, found := suite.chainA.App.GetIBCKeeper().ChannelKeeper.GetPruningSequenceEnd(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+				suite.Require().True(found)
+
+				suite.Require().Equal(uint64(1), start)
+				suite.Require().Equal(uint64(1), end)
+
+				// We expect 0 to be pruned and 0 left.
+				suite.Require().Equal(uint64(0), pruned)
+				suite.Require().Equal(uint64(0), left)
+			},
+			nil,
+		},
+		{
+			"success: stale packet state pruned up to limit",
+			func() {
+				// Send 10 packets from B -> A, creating 10 packet receipts and 10 packet acks on A.
+				suite.sendMockPackets(path.EndpointB, path.EndpointA, 10)
+			},
+			func() {},
+			func(pruned, left uint64) {
+				sequenceEnd, found := suite.chainA.App.GetIBCKeeper().ChannelKeeper.GetPruningSequenceEnd(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+				suite.Require().True(found)
+
+				// We expect nothing to be left and sequenceStart == sequenceEnd.
+				postPruneExpState(0, 0, sequenceEnd)
+
+				// We expect 10 to be pruned and 0 left.
+				suite.Require().Equal(uint64(10), pruned)
+				suite.Require().Equal(uint64(0), left)
+			},
+			nil,
+		},
+		{
+			"success: stale packet state partially pruned",
+			func() {
+				// Send 10 packets from B -> A, creating 10 packet receipts and 10 packet acks on A.
+				suite.sendMockPackets(path.EndpointB, path.EndpointA, 10)
+			},
+			func() {
+				// Prune only 5 packet acks.
+				limit = 5
```

Comment:
I think this test should set a limit of anything other than 5 so pruned/left are not both also equal to 5.